### PR TITLE
Add supply chain app

### PR DESCRIPTION
**Pull request by nikitayutanov**

### CODE REVIEW

File: apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { isValidHex } from 'utils';
+import commonStyles from '../Create.module.scss';
+import styles from './AddressForm.module.scss';
+
+type Props = {
+  id: string;
+  onSubmit: (value: { nftProgramId: Hex; ftProgramId: Hex }) => void;
+};
+
+const initialValues = {
```
Comment (Pinkdesu) [inline apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx:15]: Move it in consts, helpers

File: apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+function AddressForm({ id, onSubmit }: Props) {
+  const { api } = useApi();
+
+  const form = useForm({ initialValues, validate });
+  const { getInputProps, setFieldError } = form;
+
+  const nftInputClassName = clsx(commonStyles.input, styles.nftInput);
+
+  const handleSubmit = form.onSubmit(async ({ nftProgramId, ftProgramId }) => {
+    const isNftExists = await api.program.exists(nftProgramId);
```
Comment (Pinkdesu) [inline apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx:30]: use Promise.all

File: apps/supply-chain/src/pages/home/program/Program.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                        action={getAction(action)}
+                        label={getLabel(action)}
+                        name={getName(action)}
+                        onSubmit={handleSubmit}
+                      />
+                    )}
+                  </>
+                ) : (
+                  <SelectText value={selectTextValue} />
+                )}
```
Comment (Pinkdesu) [inline apps/supply-chain/src/pages/home/program/Program.tsx:83]: Maybe move part of the logic into a separate component, difficult to read

File: apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx (hunk 1 above)
Comment (nikitayutanov) [reply apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx:15]: I prefer to not separate small encapsulated values

Comment (nikitayutanov) [reply apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx]: Placeholder values are gone. 
As for utils, I prefer to not separate small encapsulated values

File: apps/supply-chain/src/pages/home/program/Program.tsx (hunk 3 above)
Comment (nikitayutanov) [reply apps/supply-chain/src/pages/home/program/Program.tsx:83]: Suggest what component and what logic should be inside.
I agree that the way it is rn is a bit complicated, but don't see better solution

File: apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx (hunk 2 above)
Comment (nikitayutanov) [reply apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx:27]: There's a need to get different errors if either `nft` or `ft` program doesn't exist.

Comment (nikitayutanov) [reply apps/supply-chain/src/pages/home/create/address-form/AddressForm.tsx:27]: Idk, current solution looks clearer to me:
image

**Review by Pinkdesu — APPROVED** [review]

**Pull request merged**